weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/cost/costUtils.ts

```diff
@@ -87,22 +88,45 @@ export const formatTokenCost = (cost: number): string => {
   return `$${cost.toFixed(2)}`;
 };
 
-// TODO(Josiah): this is here because sometimes the cost query is not returning all the ids I believe for unfinished calls,
-// to get this cost uptake out, this function can be removed, once that is fixed
 export const addCostsToCallResults = (
   callResults: CallSchema[],
   costResults: CallSchema[]
-) => {
-  const costDict = costResults.reduce((acc, cost) => {
-    if (cost.callId) {
-      acc[cost.callId] = cost;
+): CallSchema[] => {
+  const costDict = costResults.reduce((acc, costResult) => {
+    if (costResult.callId) {
```

Comment:
if you wanted you could make this accumulator simpler 

```
acc[call.id] = {
  costs: costResult.traceCall?.summary?.weave?.costs,
  usage: costResult.traceCall?.summary?.usage,
  }
```
then make the code below

```
const merged = {
        ...call,
        traceCall: {
          ...call.traceCall,
          summary: {
            ...call.traceCall?.summary,
            weave: {
              ...call.traceCall?.summary?.weave,
              costs: costDict[call.callId].costs,
            },
          },
          usage: costDict[call.callId].usage,
        },
      };
```

but this looks right